adapters/beachfront/beachfront.go

```diff
@@ -568,29 +563,51 @@ func (a *BeachfrontAdapter) MakeBids(internalRequest *openrtb2.BidRequest, exter
 	return bidResponse, errs
 }
 
-func setBidFloor(ext *openrtb_ext.ExtImpBeachfront, imp *openrtb2.Imp) error {
-	var floor float64
+func setBidFloor(ext *openrtb_ext.ExtImpBeachfront, imp *openrtb2.Imp, reqInfo *adapters.ExtraRequestInfo) (bool, error) {
+	var initialImpBidfloor float64 = imp.BidFloor
+	var err error
 
-	if imp.BidFloor > 0 {
-		floor = imp.BidFloor
-	} else if ext.BidFloor > 0 {
-		floor = ext.BidFloor
-	} else {
-		floor = minBidFloor
-	}
+	if imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" && imp.BidFloor > 0 {
+		imp.BidFloor, err = reqInfo.ConvertCurrency(imp.BidFloor, imp.BidFloorCur, "USD")
+
+		var convertedFromCurrency = imp.BidFloorCur
+		imp.BidFloorCur = "USD"
 
-	if floor >= 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" {
-		return &errortypes.BadInput{
-			Message: fmt.Sprintf("unsupported bid currency, %s. bids are currently accepted in USD only.", imp.BidFloorCur),
+		if err != nil {
+			if ext.BidFloor > minBidFloor {
+				imp.BidFloor = ext.BidFloor
+				return false, &errortypes.Warning{
+					Message: fmt.Sprintf("The following error was recieved from the currency converter while attempting to convert the imp.bidfloor value of %.2f from %s to USD:\n%s\nThe provided value of imp.ext.beachfront.bidfloor, %.2f USD is being used as a fallback.",
+						initialImpBidfloor,
+						convertedFromCurrency,
+						err,
+						ext.BidFloor,
+					),
+				}
+			} else {
+				return true, &errortypes.BadInput{
+					Message: fmt.Sprintf("The following error was recieved from the currency converter while attempting to convert the imp.bidfloor value of %.2f from %s to USD:\n%s\nA value of imp.ext.beachfront.bidfloor was not provided. The bid is being skipped.",
+						initialImpBidfloor,
+						convertedFromCurrency,
+						err,
+					),
+				}
+			}
 		}
 	}
 
-	if floor <= minBidFloor {
-		floor = 0
+	if imp.BidFloor < ext.BidFloor {
```

Comment:
When execution gets here, there's a possibility that `imp.BidFloor` has already been converted from a foreign currency to "USD". Are we positive that `ext.BidFloor` comes in "USD" and not the same foreign currency `imp.BidFloor` was? 

For instance: say a given request comes with both `imp.BidFloor` and `ext.BidFloor` set to "EUR". The comparisson here would be misleading because we could be comparing `imp.BidFloor` representing a price in US dollars versus `ext.BidFloor` representing Euros. Is this a feasible possibility? Can `ext.BidFloor` come in a non "USD" currency?

Reply:
@guscarreon 
ext.bidfloor is always is USD. So in the case where imp.bidfloor has been converted to USD from EUR, the ext.bidfloor is not converted (it is never converted). Note the comparison at 577. This also assumes that ext.bidfloor is USD, as is minBidFloor (which beachfront may one day decide should be more than $0.01)

If ext.bidfloor is erroneously sent in non-USD, that is the value that is used. If imp.bidfloorcur = INR, imp.bidfloor= 375.23, and ext.bidfloor = 310.21, then ext.bidfloor will be used, as the converted value of imp.bidfloor = 5.00 USD is less than ext.bidfloor = 310.21 USD, which was entered in INR due to user error, and someone's not getting any ads. I verified this with beachfront, and I agree, it's a bit counter intuitive, but it is correct.

I need to update the docs as they still say the converter doesn't work, but this bit is in there as of my previous PR. I'll push that in a couple minutes.

Reply:
Thanks for clarifying @muncha